star_tides/core/data/contact_data.py

```diff
@@ -0,0 +1,38 @@
+'''star_tides.core.data.contact_data
+'''
+
+
+from star_tides.services.databases.mongo.models.contact_model import ContactModel
+from star_tides.services.databases.mongo.models.availability import Availability
+from star_tides.core.data.location_data import LocationData
+from star_tides.core.data.engagement_data import EngagementData
+from typing import List, NamedTuple, Optional
+
+
+class ContactData(NamedTuple):
+    '''ContactData represents a Contact object's data in our system.'''
+    # Fields with NO default values.
+    name: str
+    location: LocationData
+    availability: Availability
+    engagement: EngagementData
+    # Fields with default values.
+    id: Optional[str] = None
+    user_id: Optional[str] = None
+    email: Optional[str] = None
+    phone_number: Optional[str] = None
+    job_title: Optional[str] = None
+    website_url: Optional[str] = None
+    languages: List[str] = []
+    statuses: List[str] = []
```

Comment:
Shouldn't either using the models or marshmallow eliminate the need for declaring effectively a second model?

Reply:
IMO, Models should only be used for direct reads/writes to the DB (never passed around between Actions). Schemas should only be used for parsing JSON and Model dumps. I don't think using Marshmallow everywhere is a good idea since it only works in terms of dictionaries, which don't have explicit fields or types (bug prone IMO). Hence, we have these Data classes that are typesafe and are made explicitly for passing between Actions (and Controllers). What do you think of this? My opinion is heavily influenced by https://developers.google.com/protocol-buffers. If we used these, we would have the benefits of type safety as well as nice JSON (de)serialization.